docs/en/ingest-management/elastic-agent/scaling-on-kubernetes.asciidoc

```diff
@@ -0,0 +1,272 @@
+
+
+[discrete]
+=== Specifying resources and limits in Agent manifests
+
+Based on our tests we advise to configure only the `limit` section of the `resources` section in the manifest. In this way the `request`'s settings of the `resources` will fall back to the `limits` specified. The `limits` is the upper bound limit of your microservice process, meaning that can operate in less resources and protect Kubernetes to assign bigger usage and protect from possible resource exhaustion.
+
+[source,yaml]
+------------------------------------------------
+resources:
+    limits:
+      cpu: "1000m"
+      memory: "200Mi"
+------------------------------------------------
+
+
+Based on our https://github.com/elastic/elastic-agent/blob/7467fd72bccab9a0e1b0adc1761cde8520390943/docs/elastic-agent-scaling-tests.md[tests], the following table provides guidelines to adjust {agent} limits on different kubernetes sizes:
+
+Sample Elastic Agent Configurations:
+|===
+| No of Pods in K8s Cluster | Leader Agent Resources | Rest of Agents 
+| 1000   | cpu: "1500m",  memory: "800Mi" | cpu: "300m",  memory: "600Mi" 
+| 3000   | cpu: "2000m",  memory: "1500Mi" | cpu: "400m",  memory: "800Mi" 
+| 5000   | cpu: "3000m",  memory: "2500Mi" | cpu: "500m",  memory: "900Mi" 
+| 10000  | cpu: "3000m",  memory: "3600Mi" | cpu: "700m",  memory: "1000Mi" 
+|===
+
+> Above tests run with {agent} version 8.7 and scraping period of 10sec in Kubernetes Policy. Those numbers are just indicators and should be validated per different kubernetes policy configuration, along with applications that the kubernetes cluster might include
+
+[discrete]
+=== Proposed Agent Installations for large scale
+
+Although daemonset installation is simpler, it can not differentiate the alternative agent resource requirements depending on the collected metrics. The need for better 
+resource assignment led us to alternatives installation methods. 
+
+IMPORTANT: The main suggestion for big scale clusters *is to install Elastic Agent as side container with Kube State Metrics Shard*. The installation is explained in details https://github.com/elastic/elastic-agent/tree/909b09def863205ae928f440d04851494c8e9933/docs/manifests/kustomize-autosharding[here]
+
+In general **3 alternative configuration methods** to deploy elastic-agent in big scale Kubernetes clusters with Kube State Metrics (KSM) in sharding configuration have been verified:
+
+1. With `hostNetwork:false` 
+  - Elastic Agent as Side Container within KSM Shard pod
+  - For non-leader Elastic Agent deployments that collect per KSM shards
+2. With `podAntiAffinity` to isolate the Elastic Agent daemonset pods from rest of deployments
+3. With `taint/tolerations` to isolate the Elastic Agent daemonset pods from rest of deployments
```

Comment:
I'm unclear if these 2 and 3 even work without resulting in pods stuck in pending state.

Reply:
Let me provide some examples for 2 and 3 and maybe we can remove the 2 ?

Reply:
So I have repeated the tests and added examples here https://github.com/elastic/elastic-agent/blob/0ea3b3ed1da29d47dcd707a75cf837c9be07ba08/docs/elastic-agent-ksm-sharding.md


For 2 Affinity:
Indeed the pod will **remain** in Pending state. The antiaffinity mather is `requiredDuringSchedulingIgnoredDuringExecution:`, so it will schedule it but will ignore the the Exucution,thus it will remain in pending state

```bash
❯ kgp -A | grep elastic
kube-system   elastic-agent-64ctk                                              1/1     Running   0          4m44s
kube-system   elastic-agent-8db69556b-6t4qv                                    1/1     Running   0          4m48s
kube-system   elastic-agent-cj2zk                                              1/1     Running   0          4m44s
kube-system   elastic-agent-rlnqk                                              1/1     Running   0          4m44s
kube-system   elastic-agent-tt5lg                                              1/1     Running   0          4m44s
kube-system   elastic-agent-txxfp                                              0/1     Pending   0          4m44s
```

So I agree, do you think shall we remove completely the antiafinity configuation?

For 3 Tolerations:
The pod you define as taint is not available at all. So we have no pods in pending state

So it looks like:

```bash
kubectl get pods -n kube-system| grep elastic
elastic-agent-2jw86                                              1/1     Running   0          4m33s
elastic-agent-56cfc5759-7bmz7                                    1/1     Running   0          4m11s     < This is the deployment Agent
elastic-agent-65f59                                              1/1     Running   0          4m33s
elastic-agent-8j9t6                                              1/1     Running   0          4m33s
elastic-agent-b5mgk                                              1/1     Running   0          4m33s
elastic-agent-fn5tt                                              1/1     Running   0          4m33s
elastic-agent-g4m92                                              1/1     Running   0          4m33s
elastic-agent-h4wbc                                              1/1     Running   0          4m33s
```

So I guess we can keep only taint as more complex example. Lets agree on tomorrow sync and I will update the docs

Reply:
@pjbertels I have removed affinity configurations from all docs